### PR TITLE
ENTERPRISE-61: Add Jenkins integration build faild documentation

### PR DESCRIPTION
**Pull request by vinokurig**
Add Jenkins integration build faild documentation

### CODE REVIEW

File: codenvy-docs.iml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bmicklea) [inline codenvy-docs.iml]: Why is this file needed?  What does it do?

Comment (eivantsov) [inline codenvy-docs.iml]: Oops. IntelliJ file. Needs to be removed.

File: src/main/_docs/integration-guide/integration-continuous-integration.md [hunk 2]
```diff
@@ -52,66 +67,68 @@ You need a Codenvy Factory configured to use the project you want associated wit
...
 
-#### Git
-**For GitHub**
-Update the `codenvy.env` with `GitHub` webhooks properties. Note that you can rename "WEBHOOK1" with any identifier. Webhooks are tied to connectors through Factory IDs. There can be multiple webhooks (if you need to receive webhooks from multiple repositories), i.e. a new group of webhook properties can be created. In this case, change identifier WEBHOOKID and FACTORY1_ID, which can be any strings.
+### Git
+The Jenkins integration supports both GitHub and BitBucket Server repos.
+
+#### For GitHub
+
+Update the `codenvy.env` with `GitHub` webhooks properties. Note that you can rename "WEBHOOK1" with any identifier. Webhooks are tied to connectors through Factory IDs. There can be multiple webhooks (if you need to receive webhooks from multiple repositories) - in this case, add a new entry with a different name for WEBHOOKID and FACTORY1_ID.
```
Comment (vinokurig) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: >in this case, add a new entry with a different name for WEBHOOKID and FACTORY1_ID.

I would rephrase this:
in this case, add a new group of entries with a different name for WEBHOOKID.

FACTORY1_ID is described below

File: src/main/_docs/integration-guide/integration-continuous-integration.md [hunk 3]
```diff
@@ -52,66 +67,68 @@ You need a Codenvy Factory configured to use the project you want associated wit
...
 
 ```text  
 CODENVY_GITHUB_WEBHOOK_WEBHOOK1_REPOSITORY_URL=https://github.com/example/testrepo.git
 CODENVY_GITHUB_WEBHOOK_WEBHOOK1_FACTORY1_ID=factory1Id
 ```
 
-One webhook can update more than one Factory which can be added to configuration with a different Factory identifier, for example `FACTORY2_ID`:
+One webhook can update more than one Factory. Add a different Factory identifier, for example `FACTORY2_ID`:
```
Comment (vinokurig) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: I would rephrase this:
Each webhook can update more than one Factory - in this case, add a new entry with a FACTORY<N>_ID, where N is incremented number of the factory in the current group of entries, for example ...

the same for Bitbucket Server webhooks

File: src/main/_docs/integration-guide/integration-continuous-integration.md [hunk 4]
```diff
@@ -52,66 +67,68 @@ You need a Codenvy Factory configured to use the project you want associated wit
...
 ```
-In this case, webhook with ID `WEBHOOK_WEBHOOK1` will update 2 Factories - `FACTORY1_ID` and `FACTORY2_ID`.
 
-**For BitBucket Server**
-Update the `codenvy.env` with `Bitbucket Server` webhooks properties. Webhooks are tied to connectors through Factory IDs. There can be multiple webhooks, i.e. a new group of webhook properties can be created. In this case, change identifier WEBHOOKID, which can be any string.
+In this case, the webhook with ID `WEBHOOK_WEBHOOK1` will update two Factories - `FACTORY1_ID` and `FACTORY2_ID`.
+
+#### BitBucket Server
+
+Update the `codenvy.env` with `Bitbucket Server` webhooks properties. Webhooks are tied to connectors through Factory IDs. There can be multiple webhooks (if you need to receive webhooks from multiple repositories) - in this case, add a new entry with a different name for WEBHOOKID and FACTORY1_ID.
```
Comment (vinokurig) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: FACTORY1_ID is not needed here, see comment above

**Comment by vinokurig** [comment]
@bmicklea please review my comments

File: src/main/_docs/integration-guide/integration-continuous-integration.md [hunk 5]
```diff
@@ -7,41 +7,56 @@ permalink: /:categories/continuous-integration/
 ---
 {% include base.html %}
 
-Codenvy is connected to your repo so any change made to the repo that would normally trigger a CI job will continue to trigger a CI job when the change is made in Codenvy.
+Codenvy can use [Factories]({{base}}{{site.links["factory-getting-started"]}}) with a Jenkins CI system to generate developer workspaces pre-configured with the context of the CI job. For example, a failed CI build email can be customized to include a link to two factories that open a workspace pre-configured to:
+* The branch and commit ID that broke the build for the project.
```
Comment (bmicklea) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: The branch isn't specified in the Factory, but if the jenkins build is for a specific branch then the factory will use that branch, right?

File: src/main/_docs/integration-guide/integration-continuous-integration.md [hunk 6]
```diff
@@ -7,41 +7,56 @@ permalink: /:categories/continuous-integration/
 ---
 {% include base.html %}
 
-Codenvy is connected to your repo so any change made to the repo that would normally trigger a CI job will continue to trigger a CI job when the change is made in Codenvy.
+Codenvy can use [Factories]({{base}}{{site.links["factory-getting-started"]}}) with a Jenkins CI system to generate developer workspaces pre-configured with the context of the CI job. For example, a failed CI build email can be customized to include a link to two factories that open a workspace pre-configured to:
+* The branch and commit ID that broke the build for the project. 
+* The head of the master branch for the project.
```
Comment (bmicklea) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: That link doesn't clarify things for me. If the jenkins job is setup to build master then the head factory will open with a clone of master, won't it?  If the jenkins job is setup to build "brad" branch then the factory will open with a clone of "brad" right?

File: src/main/_docs/integration-guide/integration-continuous-integration.md (hunk 2 above)
Comment (bmicklea) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: ok

File: src/main/_docs/integration-guide/integration-continuous-integration.md (hunk 6 above)
Comment (eivantsov) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: HEAD of branch. master is a branch too. So, if it is master then it is head of master.

File: src/main/_docs/integration-guide/integration-continuous-integration.md (hunk 3 above)
Comment (bmicklea) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: Okay - I also noticed that the BitBucket Server section was using a github URL so I replaced that.

File: src/main/_docs/integration-guide/integration-continuous-integration.md (hunk 4 above)
Comment (bmicklea) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: ok

File: src/main/_docs/integration-guide/integration-continuous-integration.md (hunk 6 above)
Comment (bmicklea) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: fixed

File: src/main/_docs/integration-guide/integration-continuous-integration.md (hunk 5 above)
Comment (bmicklea) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: I think I have corrected this now, please review.

File: src/main/_docs/integration-guide/integration-continuous-integration.md (hunk 6 above)
Comment (vinokurig) [inline src/main/_docs/integration-guide/integration-continuous-integration.md]: You are right, but 
>The head of the master branch for the project.

From this user can think that the factory configured for the webhook points to master, but in fact, accordig to the Jenkins integration flow, the factory points to developer branch. May be it would be better to write:
`The head of the branch for the project.` ?
